### PR TITLE
sig-windows: create release signal dashboard on testgrid

### PR DESCRIPTION
**Pull request by chewong**
Signed-off-by: Ernest Wong <chuwon@microsoft.com>

The goal of this PR is to mimic https://testgrid.k8s.io/sig-release by removing redundant sig-windows testgrid dashboards and breaking them down into releases so we can have a clearer signal for each supported release

- removed the following testgrid dashboards:
  - sig-windows-releases
  - sig-windows-containerd
  - sig-windows-azure
- created the following testgrid dashboards:
  - sig-windows-1.18-release
  - sig-windows-1.19-release
  - sig-windows-1.20-release
  - sig-windows-master-release
  - sig-windows-containerd-hyperv
- move all presets to `presets.yaml`
- enforce presubmit job name - `pull-<cluster provisioning tool>-<what the job is testisng>-<OS>-<container runtime>`

### CODE REVIEW

**Comment by chewong** [comment]
/assign @jsturtevant @marosset 
/cc @jeremyje  
since I edited some GCE windows jobs

File: config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml:167]: Where is $(GINKGO_FOCUS_PARALLEL) being set?

Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml:167]: It's set in https://github.com/kubernetes/test-infra/pull/21457/files#diff-df283afbd02f2f70d68a1151d0b93b86f3ce1a8bf4766ef4d812aae58ac26a31R11-R18

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-sac.yaml
Comment (marosset) [inline config/jobs/kubernetes-sigs/sig-windows/sig-windows-sac.yaml:54]: I don't think we want the SAC test passes to show up in sig-windows-master-release, at least not the aks-engine ones.
Can we keep the sig-windows-sac dashboard?
Or maybe split it up into sig-windows-{sac-{versions}}?

File: config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml
Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml:167]: found it.  The file was minimized by default

File: config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml:173]: These should probably still have the version number (1.18) in the tab name?

File: config/jobs/kubernetes-sigs/sig-windows/release-1.20-windows.yaml
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/release-1.20-windows.yaml:226]: do we still want to have a sig-windows-azure for a holistic view across azure?

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml:142]: Should we keep this around for the last few images that need to be converted over?

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-sac.yaml
Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/sig-windows-sac.yaml:54]: I was thinking the same with the hyper-v tests

Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/sig-windows-sac.yaml:54]: I like keeping the `sig-windows-sac`

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml:18]: I like breaking these out so they are not duplicated across the version. Much easier to maintain. did you double check that we didn't have specific tests enable for specific releases?

If we did have a release that needed a test enabled/disabled we could do that on the test it's self?  

For instance it looks like we dropped the skip for 1.20 here: https://github.com/kubernetes/test-infra/pull/20970

**Review by jsturtevant — CHANGES_REQUESTED** [review]

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml
Comment (marosset) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml:18]: Yea! big +1 for this change!

File: config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml
Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml:173]: The reason why I didn't include the version name is that it's already encoded in the dashboard name

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml
Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml:18]: we can do something like `$(GINKGO_SKIP_SERIAL)|\[sig-scheduling\].SchedulerPredicates.\[Serial\].validates.that.there.is.no.conflict.between.pods.with.same.hostPort` for special cases but in general, most of the jobs follow the same `GINKGO_FOCUS` and `GINKGO_SKIP` across different releases

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml
Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml:142]: All required test images are already using promoter registry so we should be able to test them with our normal presubmit jobs.

File: config/jobs/kubernetes-sigs/sig-windows/release-1.20-windows.yaml
Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.20-windows.yaml:226]: I would prefer each job to appear in at most one dashboard (unless it's an informing job, in which case we would have to display it in the sig-release-informing dashboard) to minimize the number of dashboards we maintain

File: config/jobs/kubernetes-sigs/sig-windows/sig-windows-sac.yaml
Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/sig-windows-sac.yaml:54]: Sounds good, let me add it back

File: config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml
Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml:173]: looking at a sig-release dashboard I see most have the release name encoded: https://testgrid.k8s.io/sig-release-1.20-informing.  It's minor but would be helpful if it does get duplicated across dashboards

File: config/jobs/kubernetes-sigs/sig-windows/release-1.20-windows.yaml
Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.20-windows.yaml:226]: sounds good.

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml
Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml:142]: how will we test the etcd image before cutting over?

Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml:142]: Assuming that the etcd image has been promoted, we should be able to test it in the k/k PR where we bump the version/change the registry in https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go#L228.  

The reason why we had `pull-kubernetes-e2e-aks-engine-windows-containerd-test-images` in the first place is that there was a period in February where we were overwriting the `promoterRegistry` to point to our ACR due to a couple of invalid images promoted to k8s.gcr.io. There is no way to test newly promoted images because we would have to push those images to our ACR. Now that the problem has been resolved, the normal presubmit jobs should be able to test it since it uses the default `promoterRegistry` .

File: config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml
Comment (chewong) [reply config/jobs/kubernetes-sigs/sig-windows/release-1.18-windows.yaml:173]: That makes sense. Let me append the version to the end of the tab name

**Review by jsturtevant — COMMENTED** [review]
Thanks for splitting out the presets! 

/lgtm

**Comment by chewong** [comment]
/assign @feiskyer 
from release-engineering-approvers

**Review by marosset — APPROVED** [review]
/lgtm

Thanks for cleaning this up!

**Comment by chewong** [comment]
/hold
to prevent from being auto-merged for now
@feiskyer please still go ahead and review this PR. I will unhold this PR once it's approved. Thanks in advance

**Review by feiskyer — COMMENTED** [review]
/approve

**Comment by chewong** [comment]
/unhold
Thanks!

**Pull request merged**
